adobe/spectrum-css
# docs(combobox): update placeholder text

## Description

Removes the placeholder in the combobox template to eliminate concerns about accessibility (because placeholder text disappears when a value is present, it should not contain essential information or instructions). Text from the [guidelines](https://spectrum.adobe.com/page/combo-box/#Don%E2%80%99t-use-placeholder-text) explaining that placeholder text should not be used for combobox were also added to the Docs page for Combobox.

Also contains an update to the template used for the Docs page; where previously Combobox had its own .mdx page, it no longer seems necessary in comparison to other migrations that were completed after this one. Any relevant information from combobox.mdx was moved to combobox.stories.js and the comobobox.mdx file was removed. The combobox.mdx page did have the added benefit of allowing us to better control the order in which the stories appeared, but lately the norm has been that this doesn't justify a component having its own .mdx.

One thing led to another here and the Docs page was also reorganized with one story for each variant, each containing multiple states (closed, open, invalid, disabled, loading, etc.). This looks more similar to the more recently completed docs pages (for instance [Tabs](https://spectrum-css.netlify.app/preview/?path=/docs/components-tabs--docs), [Action Button](https://spectrum-css.netlify.app/preview/?path=/docs/components-action-button--docs), or [Alert Banner](https://spectrum-css.netlify.app/preview/?path=/docs/components-alert-banner--docs)).

There are two commits to make changes to the template for combobox as well:
1. Menu items were being passed into Combobox from combobox.stories.js, but weren't visible because the combobox template was displaying a Menu component with args hard-coded into the template. This was adjusted so that args from the story are now visible in Combobox's menu.
2. The value of the Textfield within Combobox was set to `selectedDay` or undefined. `selectedDay` is necessary for Date Picker, but I didn't see a reason it was needed within Combobox. I _think_ this might be a remnant from when [Combobox and Date Picker were split from the InputGroup package](https://github.com/adobe/spectrum-css/pull/1556). `selectedDay` was removed from Combobox in favor of a `value` argument.

Neither of these two commits is necessary and can be removed if there is a reason to keep the template as is.

These changes only affect documentation, so no changeset is necessary.

[CSS-41](https://jira.corp.adobe.com/browse/CSS-41)

## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Validation steps

Using the deployment preview, view the combobox component [default story](https://pr-3219--spectrum-css.netlify.app/preview/?path=/story/components-combobox--default) and [docs page](https://pr-3219--spectrum-css.netlify.app/preview/?path=/docs/components-combobox--docs) and :
- [x] Confirm that placeholder has been removed
- [x] Confirm that appropriate placeholder guidelines have been added
- [x] Confirm that no information has been lost from the Docs page in the deletion of combobox.mdx (all the same stories should be displayed), compare with the Combobox Docs page in [prod](https://spectrum-css.netlify.app/preview/?path=/docs/components-combobox--docs)
- [x] Confirm that testing coverage still looks the way that we want it to--this will have an impact on VRTs because the content has changed, but should otherwise remain the same 

Other considerations in the scope of this work:
1. I didn't display Help Text with this, but could if we think it's relevant (since we are showing the field label and talking about help text)
2. I removed the fallback Menu content from the template since it was overriding any menu content coming from args, but that meant that every story (with `isOpen: true`) needs to have a content arg, thus I made `content` a default arg.
3. Do changes made to the `value` property work? Do we want to display any value text or leave it blank? (If a menu item is selected in SWC's [implementation](https://opensource.adobe.com/spectrum-web-components/components/combobox/), you'll see that as the value in the text field)
4. The Menu content in the args differs slightly from the removed fallback content in the template (same items, but adds a divider and selection).

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:


6. If components have been modified, VRTs have been run on this branch:


## Screenshots



## To-do list
